datastore/datastore.go

```diff
@@ -185,11 +186,14 @@ func ParseKey(key string) ([]string, error) {
 }
 
 // newClient used to connect to KV Store
-func newClient(scope string, kv string, addr string, config *store.Config, cached bool) (DataStore, error) {
+func newClient(scope string, kv string, addr string, config *store.Config, cached bool, sequential bool) (DataStore, error) {
 
 	if cached && scope != LocalScope {
 		return nil, fmt.Errorf("caching supported only for scope %s", LocalScope)
 	}
+	if scope == LocalScope && !sequential {
```

Comment:
Yes you are right. I leave the decision in NewDataStore rather than in newClient, just in case  that making the decision depends on more parameters other than scope in future.